### PR TITLE
APP-1146: Publish api buf artifacts with named tag matching git tag

### PR DESCRIPTION
**Pull request by AdamMagaluk**
Fork code changes: https://github.com/bufbuild/buf-push-action/compare/main...AdamMagaluk:buf-push-action:main

I don't have 100% confidence in this and there isn't a great way to test this without pushing FYI.

### CODE REVIEW

File: .github/workflows/tag_and_release_protos.yml [hunk 1]
```diff
@@ -22,27 +22,32 @@ jobs:
...
       - name: Bump tag
+        id: bump
         uses: anothrNick/github-tag-action@1.52.0
         env:
           GITHUB_TOKEN: ${{ secrets.REPO_READ_TOKEN }}
           WITH_V: true
           DEFAULT_BUMP: patch
 
   buf-push:
+    needs: bump-tags
```
Comment (michaellee1019) [inline .github/workflows/tag_and_release_protos.yml:38]: Should be `bump-tag`

**Review by michaellee1019 — COMMENTED** [review]
Nice! you did end up forking it :) I thinking i found a bug. Otherwise worth a shot and LGTM

File: proto/viam/app/v1/app.proto [hunk 2]
```diff
@@ -161,7 +161,6 @@ message RobotPartHistoryEntry {
   RobotPart old = 4 [(tagger.v1.tags) = "bson:\"old\" json:\"old\""];
 }
 
-// Organizations
```
Comment (edaniels) [inline proto/viam/app/v1/app.proto:164]: undo?

**Review by edaniels — APPROVED** [review]
If it works it works, lgtm

File: proto/viam/app/v1/app.proto (hunk 2 above)
Comment (AdamMagaluk) [reply proto/viam/app/v1/app.proto:164]: I need a file in proto/** change for it to kick off the workflow. And this wasn't a great comment to begin with.

File: .github/workflows/tag_and_release_protos.yml (hunk 1 above)
Comment (AdamMagaluk) [reply .github/workflows/tag_and_release_protos.yml]: Good catch thanks.

**Review by edaniels — APPROVED** [review]
LGTM

**Pull request merged**